app/services/queue_manager.rb

```diff
@@ -0,0 +1,53 @@
+class QueueManager
+  attr_reader :page, :job_type
+
+  def QueueManager.push(page, job_type:)
```

Comment:
personal taste... Normally I go with `class << self ...` but it's verbose for one method (trying to break that habit). I don't really like how `def self.something` looks... would you prefer `self` - happy to change it.